File: core/src/com/alta_v2/mediator/serde/Updater.java
package com.alta_v2.mediator.serde;

import com.alta_v2.rendering.ScreenState;

/**
 * Provides the high level abstraction for update screen state.
 */
public interface Updater {

    /**
     * Initializes the updater.
     */
    void init(ScreenState screenState);

    /**
     * Updates the screen.
     *
     * @param delta - the time in seconds since the last render.
     */
    void update(float delta, ScreenState screenState);

    /**
     * Destroys the updater.
     */
    void destroy();
}

File: core/src/com/alta_v2/mediator/serde/Updater.kt
package com.alta_v2.mediator.serde

import com.alta_v2.common.Destroyable
import com.alta_v2.rendering.ScreenState

/**
 * Provides the high level abstraction for update screen state.
 */
interface Updater : Destroyable {
    /**
     * Initializes the updater.
     */
    fun init(screenState: ScreenState)

    /**
     * Updates the screen.
     *
     * @param delta - the time in seconds since the last render.
     */
    fun update(delta: Float, screenState: ScreenState)
}